test-framework/test-suites/integration/tests/add/test_add_pallet.py

```diff
@@ -400,3 +400,52 @@ def test_add_pallet_updates_url(self, host, run_pallet_isos_server, create_palle
 				'url': 'http://127.0.0.1:8000/minimal-1.0-sles12.x86_64.disk1.iso'
 			}
 		]
+
+	def test_add_pallet_output(self, host, create_pallet_isos, revert_export_stack_pallets):
+		# Add our minimal pallet
+		result = host.run(f'stack add pallet {create_pallet_isos}/minimal-1.0-sles12.x86_64.disk1.iso output-format=json')
+		assert result.rc == 0
+
+		assert json.loads(result.stdout) == [
+			{
+				'pallet': 'minimal',
+				'version': '1.0',
+				'release': 'sles12',
+				'arch': 'x86_64',
+				'os': 'sles'
+			}
+		]
+
+	def test_add_pallet_output_network(self, host, run_pallet_isos_server, revert_export_stack_pallets):
+		# Add the minimal pallet ISO from the network
+		result = host.run('stack add pallet http://127.0.0.1:8000/minimal-1.0-sles12.x86_64.disk1.iso output-format=json')
+		assert result.rc == 0
+
+		print(result.stdout)
+
+		assert json.loads(result.stdout) == [
+			{
+				'name': 'minimal',
+				'version': '1.0',
+				'release': 'sles12',
+				'arch': 'x86_64',
+				'os': 'sles'
+			}
+		]
```

Comment:
This test can be combined with `test_network_iso`, just add checking result.stdout after the line to check the rc of the add pallet command.